ipwb/util.py

```diff
@@ -336,3 +340,28 @@ def check_for_update(_):
         print("The installed version of ipwb is outdated.")
         print(f"* Installed: {current}\n* Latest:    {latest}")
         print("Please run `pip install --upgrade ipwb` to upgrade.")
+
+
+def is_wacz(path):
+    # TODO: add logic to check if wacz
+    # the py-wacz validator inherits many dependencies,
+    # so ad hoc here for now
+    return is_zipfile(path)
+
+
+def get_warc_paths_in_wacz(wacz_path):
+    with ZipFile(wacz_path) as z:
+        return [w for w in z.namelist() if w.startswith('archive/')]
+
+
+def extract_warcs_to_disk(wacz_path, warc_paths):
+    for warc in warc_paths:
+        with ZipFile(wacz_path) as z:
+            z.extract(warc)
```

Comment:
I would expect a post-indexing cleanup process where these extracted WARCs are removed after they are indexed.

Reply:
A post-indexing cleanup process has been implemented in [2566b7c](https://github.com/oduwsdl/ipwb/pull/770/commits/2566b7cfc47667c78de532ee7d9bcbed340ac16c). Thanks for the suggestion, @ibnesayeed

Reply:
I think, we should create a temporary folder and extract files into that, then remove the whole temporary folder for cleanup. Some reasons to use a temporary folder:

* The process might not have write permissions in the folder where things are extracted
* If there were WARC files with duplicate names in more than one WACZ files, they might get overwritten, and if the name conflicts with a WARC file outside of the WACZ file, that file will be deleted during cleanup, which is unintended

Reply:
Good catch. Do you suggest we rename the files to some temporary filename and put it in a temporary location? This might help to prevent conflicts.

Extract (from wacz) --> (destination):
/archive/mywarc1.warc --> /tmp/js9gar/jkldfsdf.warc
/archive/mywarc2.warc --> /tmp/js9gar/234vdsf.warc
/archive/mywarc3.warc --> /tmp/js9gar/0sdf89a.warc

Then delete files at (destination) or more simply the /tmp/js9gar directory.

Reply:
Resolved with [e4b9998](https://github.com/oduwsdl/ipwb/pull/770/commits/e4b9998f82f7cb75f63ed808ec5080c0e7449074)